### PR TITLE
Docs: Fundamentals of Block Development - block in the editor

### PR DESCRIPTION
**Pull request by juanmaguitar**
Introduces the block in the editor and serves as an entry point for the related documentation.
This PR closes the issues https://github.com/WordPress/gutenberg/issues/56273 and https://github.com/WordPress/gutenberg/issues/55946

It also incoportates the content from [Block Controls: Block Toolbar and Settings Sidebar](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/block-controls-toolbar-and-sidebar/) and removed this page from the Sidebar Menu 

@ndiego the page [Block Controls: Block Toolbar and Settings Sidebar](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/block-controls-toolbar-and-sidebar/) needs to be removed and redirected to https://developer.wordpress.org/block-editor/getting-started/fundamentals/block-in-the-editor/

### CODE REVIEW

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+- [`Panel`](https://wordpress.github.io/gutenberg/?path=/docs/components-panel--docs)
+- [`ToggleControl`](https://wordpress.github.io/gutenberg/?path=/docs/components-togglecontrol--docs)
+- [`ExternalLink`](https://wordpress.github.io/gutenberg/?path=/docs/components-externallink--docs)
+
+The package [`@wordpress/block-editor`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/) includes a library of components and hooks for the Block Editor, including those to define custom settings controls for the block in the Editor. Some of the components most commonly used from this package are:
+- `RichText`
+- `BlockControls`
+- `InspectorControls`
+- `InnerBlocks`
+- `PanelColorSettings`
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]: Should these link to the respective sections of the Reference guide? For example: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/#richtext

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+- [`isSelected`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#isselected) - boolean that communicates whether the block is currently selected
+
+WordPress provides a lot of built-in standard components that can be used to define the interface of the block in the editor. These built-in components are available via packages such as [`@wordpress/components`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-components/) or [`@wordpress/block-editor`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/).
+
+<div class="callout">
+The WordPress Gutenberg project uses <a href="https://wordpress.github.io/gutenberg/?path=/docs/docs-introduction--page">Storybook</a> to document the UI components available from WordPress packages.
+</div>
+
+Custom settings controls for the block in the `Block Toolbar` or in the `Settings Sidebar` can also be defined through this `Edit` React component via built-in components such as:
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
Custom settings controls for the block in the Block Toolbar or the Settings Sidebar can also be defined through this `Edit` React component via built-in components such as:
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+- `InspectorControls`
+- `InnerBlocks`
+- `PanelColorSettings`
+
+<div class="callout callout-tip">
+The package <a href="https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/"><code>@wordpress/block-editor</code></a> also provide the tools to create and use standalone block editors.
+</div>
+
+A good workflow when using a component for the Block Editor is:
+- Import the component from a Wordpress package
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
- Import the component from a WordPress package
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+- Include the corresponding code for the component into the JSX code
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
- Add the corresponding code for the component to your project in JSX format
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+- If needed, define `attributes` in `block.json` and define event handlers to update those attributes with `setAttributes`
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
- Most built-in components will be used to set [block attributes](https://developer.wordpress.org/block-editor/getting-started/fundamentals/block-json/#using-attributes-to-store-block-data), so define any necessary attributes in `block.json` and create event handlers to update those attributes with `setAttributes` in your component
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+- If needed, adapt the code to be serialized and stored in the database
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md:43]: I don't fully understand this line 😅

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+The `props` object received by the block's `Edit` React component includes:
+- [`attributes`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#attributes) - attributes object
+- [`setAttributes`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#setattributes) - method to update the attributes object
+- [`isSelected`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#isselected) - boolean that communicates whether the block is currently selected
+
+WordPress provides a lot of built-in standard components that can be used to define the interface of the block in the editor. These built-in components are available via packages such as [`@wordpress/components`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-components/) or [`@wordpress/block-editor`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/).
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
WordPress provides many built-in standard components that can be used to define the interface of the block in the editor. These built-in components are available via packages such as [`@wordpress/components`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-components/) and [`@wordpress/block-editor`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/).
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+<div class="callout">
+The WordPress Gutenberg project uses <a href="https://wordpress.github.io/gutenberg/?path=/docs/docs-introduction--page">Storybook</a> to document the UI components available from WordPress packages.
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
The WordPress Gutenberg project uses <a href="https://wordpress.github.io/gutenberg/?path=/docs/docs-introduction--page">Storybook</a> to document the user interface components that are available in WordPress packages.
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+				tagName="p"
+				onChange={ onChangeContent }
+				value={ attr.content }
+			/>
+		</div>
+	);
+}
+```
+
+_See the [full block example](https://github.com/WordPress/block-development-examples/tree/trunk/plugins/block-toolbar-ab967f) of the [code above](https://github.com/WordPress/block-development-examples/blob/trunk/plugins/block-toolbar-ab967f/src/edit.js)_
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
_See the [full block example](https://github.com/WordPress/block-development-examples/tree/trunk/plugins/block-toolbar-ab967f) of the [code above](https://github.com/WordPress/block-development-examples/blob/trunk/plugins/block-toolbar-ab967f/src/edit.js)._
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+				style={ {
+					backgroundColor: attributes.bg_color,
+					color: attributes.text_color,
+				} }
+			/>
+		</div>
+	);
+}
+```
+_See the [full block example](https://github.com/WordPress/block-development-examples/tree/trunk/plugins/settings-sidebar-82c525) of the [code above](https://github.com/WordPress/block-development-examples/blob/trunk/plugins/settings-sidebar-82c525/src/edit.js)_
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
_See the [full block example](https://github.com/WordPress/block-development-examples/tree/trunk/plugins/settings-sidebar-82c525) of the [code above](https://github.com/WordPress/block-development-examples/blob/trunk/plugins/settings-sidebar-82c525/src/edit.js)._
```

File: docs/getting-started/fundamentals/block-in-the-editor.md [hunk 11]
```diff
@@ -0,0 +1,168 @@
+# The block in the Editor
+
+The Block Editor is a React Single Page Application (SPA) and every block in the editor is displayed through a React component defined in the `Edit` property of the settings object used to [register the block on the client](https://developer.wordpress.org/block-editor/getting-started/fundamentals/registration-of-a-block/#registration-of-the-block-with-javascript-client-side).
```
Comment (ndiego) [inline docs/getting-started/fundamentals/block-in-the-editor.md]:
```suggestion
The Block Editor is a React Single Page Application (SPA) and every block in the editor is displayed through a React component defined in the `edit` property of the settings object used to [register the block on the client](https://developer.wordpress.org/block-editor/getting-started/fundamentals/registration-of-a-block/#registration-of-the-block-with-javascript-client-side). 
```
I think this should be `edit` 🤔 The `Edit` component is assigned to the `edit` property.

**Review by ndiego — CHANGES_REQUESTED** [review]

File: docs/getting-started/fundamentals/block-in-the-editor.md (hunk 1 above)
Comment (juanmaguitar) [reply docs/getting-started/fundamentals/block-in-the-editor.md]: You're right. I have added proper links when available 👍

**Comment by juanmaguitar** [comment]
Thanks @ndiego for the review and the suggestions.
I have accepted them and added a few more updates based on your feedback,

**Review by ndiego — APPROVED** [review]
🚢

**Pull request merged**